File: common/src/main/java/de/hype/bbsentials/shared/objects/ChChestData.java
package de.hype.bbsentials.shared.objects;

import de.hype.bbsentials.shared.constants.ChChestItem;

import java.util.List;

public class ChChestData {
    public String finder;
    public Position coords;
    public List<ChChestItem> items;

    public ChChestData(String finder, Position coords, List<ChChestItem> items) {
        this.finder = finder;
        this.coords = coords;
        this.items = items;
    }

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof ChestLobbyData)) return false;
        return ((ChChestData) obj).coords.equals(coords);
    }

    @Override
    public int hashCode() {
        return coords.hashCode();
    }
}

File: common/src/main/java/de/hype/bingonet/shared/objects/ChChestData.kt
package de.hype.bingonet.shared.objects

import de.hype.bingonet.shared.constants.ChChestItem

open class ChChestData(
    @JvmField var finder: String,
    @JvmField var coords: Position,
    @JvmField var items: MutableList<ChChestItem>
) {
    override fun equals(obj: Any?): Boolean {
        if (obj !is ChChestData) return false
        return obj.coords == coords
    }

    override fun hashCode(): Int {
        return coords.hashCode()
    }
}
